docs/docs/guides/troubleshooting.mdx

```diff
@@ -155,6 +155,72 @@ LogBox.ignoreLogs([
 
 See [the accessibility overview](accessibility) to learn more about Reduced Motion.
 
+### Tried to modify key of an object which has been converted to a Shareable.
+
+**Problem:** This warning is displayed to inform the user that a Shared Value should be used or an object used in a worklet should be accessed more granularly.
+
+#### 1. Not using Shared Values.
+
+You might get this warning when you do something along the lines of:
+
+```js
+const obj = { prop: 1 };
+function worklet() {
+  'worklet';
+  console.log(obj.prop);
+}
+runOnUI(worklet)();
+obj.prop = 2; // Warning will be displayed here.
+runOnUI(worklet)();
+```
+
+and expect the results to be `1` and `2`. However, the results will be `1` and `1` because `obj` is not a Shared Value and is only copied to UI runtime once. Therefore, in development builds, we make the object immutable and add this warning after copying it to signal that it's not a valid use of Reanimated. To fix this, you should use a Shared Value instead:
+
+**Solution:**
+
+```js
+const obj = useSharedValue({ prop: 1 });
```

Comment:
Can we rename obj to sv? Also, solution should be a diff, what do you think?

Reply:
Yeah, diff is a nice idea here.